src/plugins/intel_cpu/src/nodes/qkv_proj.cpp

```diff
@@ -277,22 +411,27 @@ bool QKVProjection::isSupportedOperation(const std::shared_ptr<const ov::Node>&
                     return false;
                 }
             }
-            auto proj_pshape1 = op->input_value(1).get_shape();
-            auto proj_pshape2 = op->input_value(2).get_shape();
-            auto proj_pshape3 = op->input_value(3).get_shape();
-            if ((proj_pshape1[1] % CACHE_BLK_K_SIZE) != 0) {
+            const auto& config = node_qkv->get_config();
+            if ((config.hidden_size % CACHE_BLK_K_SIZE) != 0) {
                 errorMessage = "QKVProjection input channel size is not multiple of cache blocking size";
                 return false;
             }
-            if ((proj_pshape1[0] % REG_BLK_K_SIZE) != 0) {
+
+            if (config.quantized && (fcDynamicQuantizationGroupSize < static_cast<uint64_t>(config.hidden_size))) {
```

Comment:
Just for note: if the model is weight compressed but dynamic quantization not enabled, this check will force to the non-fused path.